modules/optimization/src/reporters/ParsedScalars.C

```diff
@@ -0,0 +1,129 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ParsedScalars.h"
+
+registerMooseObject("OptimizationApp", ParsedScalars);
+
+InputParameters
+ParsedScalars::validParams()
+{
+  InputParameters params = GeneralReporter::validParams();
+  params += FunctionParserUtils<false>::validParams();
+  params.addClassDescription("Apply parsed functions to scalar entries held in reporters.");
+  params.addParam<std::string>("name", "result", "Name of output reporter.");
+  params.addRequiredCustomTypeParam<std::string>(
+      "expression", "FunctionExpression", "function expression");
+  params.addRequiredParam<std::vector<ReporterName>>("reporter_names", "Reporter names ");
+  params.addRequiredParam<std::vector<std::string>>("reporter_symbols",
+                                                    "Expression symbol for each reporter");
+  params.addParam<std::vector<std::string>>(
+      "constant_names",
+      {},
+      "Vector of constants used in the parsed function (use this for kB etc.)");
+  params.addParam<std::vector<std::string>>(
+      "constant_expressions",
+      {},
+      "Vector of values for the constants in constant_names (can be an FParser expression)");
+  params.addParam<bool>(
+      "use_t", false, "Make time (t) variables available in the function expression.");
+
+  // This reporters is for postprocessing optimization results and shold be exectuted at the end of
+  // execution
+  params.set<ExecFlagEnum>("execute_on") = EXEC_TIMESTEP_END;
+  return params;
+}
+
+ParsedScalars::ParsedScalars(const InputParameters & parameters)
+  : GeneralReporter(parameters),
+    FunctionParserUtils(parameters),
+    _use_t(getParam<bool>("use_t")),
+    _output_reporter(
+        declareValueByName<Real>(getParam<std::string>("name"), REPORTER_MODE_REPLICATED))
+{
+  // Get symbols to corresponding reporter names
+  // need symbols because reporter names have a "/" and that will not feed into fparser
+  const std::vector<ReporterName> reporter_names(
+      getParam<std::vector<ReporterName>>("reporter_names"));
+  const std::vector<std::string> reporter_symbols(
+      getParam<std::vector<std::string>>("reporter_symbols"));
+
+  if (reporter_names.size() != reporter_symbols.size())
```

Comment:
I'll add a base class for ParsedScalars and ParsedVectors to sets up the common part in the constructor.

Reply:
You should append Reporter to the name so there's no overlap

Reply:
@GiudGiud Will you take another look at the code, input files and naming.  I wanted to get that all correct before I fixed the documentation.